hack/resolve-identityhashes.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"/..
+
+if [[ -z ${2} ]]
+then
+  echo "Resolve identityhashes placeholders "
+  echo "     resolve-identityhashes.sh <kcp-stable|kcp-unstable> file-to-resolve"
```

Comment:
I saw your PR in the application-service, and it's nice that it's included in the pipeline, but it makes any local changes (for dev/testing purposes) in the infra-deployments very hard.

My thought was that there would be a script executable directly from the infra-deployments for example:
```
generate-apiexport-overlays.sh ./components/application-service
```
that would take the path to the application folder and:
1. get the APIExport from the application-service directory by calling `kubectl kustomize <provided-path>/base` and extract the actual APIExport content by some fancy `sed` command
2. store the copy of the APIExport in `/dev` overlay
3. replace placeholders with kcp-stable identity hashes and store the modified APIExport in `/kcp-stable` overlay
4. replace placeholders with kcp-unstable identity hashes and store the modified APIExport in `/kcp-unstable` overlay

which means all the steps at once. That would make sure that everything is in sync even if it's called "locally". The same script could be used also in PR checks. And you could also run the same in the pipeline.

WDYT?

Reply:
Yop, that could work, I wanted to avoid using external tools (kubectl/kustomize) since they are not part of the image for the CI (https://github.com/redhat-appstudio/build-definitions/blob/main/tasks/update-infra-deployments.yaml#L17), but that could be updated.

Reply:
@MatousJobanek  Script rewritten to your proposal, https://github.com/redhat-appstudio/build-definitions/pull/212 created.

Reply:
Usage example -> https://github.com/redhat-appstudio/build-service/pull/65 (still need some updates here in infra-deployments after these changes are merged)

Reply:
Added generated resources by running script on https://github.com/redhat-appstudio/build-service/pull/65